### DISCUSSION TITLE
SmartCookieWeb Preview: Needed fixes and potential new features

### DISCUSSION THREAD

**Discussion by Bardock88** [CookieJarApps/SmartCookieWeb-Preview#122] [Ideas]
SmartCookieWeb Preview: Current version ⬇️
![Screenshot_20230621-203424](https://github.com/CookieJarApps/SmartCookieWeb-Preview/assets/102046116/62f050ee-fc55-47e3-8d18-89a0e82aa39b)

Need to fix:
1° Can't get into 'Downloads' when it's selected from the three-dot menu on the homepage:

https://github.com/CookieJarApps/SmartCookieWeb-Preview/assets/102046116/31b06329-304f-4cff-b058-339dfd8258e5

2° When bookmarking a site and adding it again the same site is duplicated instead of showing a popup saying the site has already been added:

https://github.com/CookieJarApps/SmartCookieWeb-Preview/assets/102046116/4a55a0ef-4b69-4093-b888-6969a47d9523

3° The 'About:About' page does not properly scale to mobile resolution =>

Wrong appearance:
![Screenshot_20230618-213753](https://github.com/CookieJarApps/SmartCookieWeb-Preview/assets/102046116/01160eb0-7aa7-4c8e-a55c-45496f8f9083)

Right appearance:
![Screenshot_20230618-213813](https://github.com/CookieJarApps/SmartCookieWeb-Preview/assets/102046116/ba61d12c-a840-4a4b-a7d2-8116209af540)

4° Reorder options menu list when on homepage to 'history', 'bookmarks', 'downloads', 'extensions' and 'settings':
![Screenshot_20230609-205840_1](https://github.com/CookieJarApps/SmartCookieWeb-Preview/assets/102046116/9dbbe805-0a4e-4b0c-a27b-04505ff486d4)

5° Reorder option menu list when accessing a website to 'New tab', 'New private tab', 'Add to home screen', 'Save to PDF', 'Print', 'History', 'Bookmarks', ' Downloads', 'Extensions', 'Desktop site', 'Find in page' and 'Settings':
![Screenshot_20230616-223631_1](https://github.com/CookieJarApps/SmartCookieWeb-Preview/assets/102046116/f26731fc-cd36-42f5-a662-db19543bd719)

Possible resources:
1° 'Downloads' button in the three-dot menu when the site is open in a normal or private tab

2° The possibility to select several bookmarks to move at once to a certain folder as in browsers based on Blink.

3° Option in 'Bookmarks' menu to open bookmark in private tab and open in private tab in background.

4° Option to choose which installed extensions can be added to the toolbar.

5° Include the 'FF Wallpapers' or allow the user to customize the wallpaper to their liking:
![Screenshot_20230618-220011](https://github.com/CookieJarApps/SmartCookieWeb-Preview/assets/102046116/53d363b8-3ae0-4573-a40b-9b02d0be206f)

6° Increase the display of the list of extensions from 50 to 100 or an undetermined value when using a custom collection of extensions.

7° Feature of holding the 'Back' button on Android to show the submenu of previously accessed sites, when you have a site open as in FF:

https://github.com/CookieJarApps/SmartCookieWeb-Preview/assets/102046116/0bb130fe-5b7a-42e4-a741-ab5224fa51fd

8° Add the 'undo tab closing' function:

https://github.com/CookieJarApps/SmartCookieWeb-Preview/assets/102046116/716f9a26-0192-401b-beb7-20acc74ad863

9° Include 'Exit' button in the three-dot menu.

At the moment, these are aspects that I have observed and that can improve the project even more and if I forgot something important, please let me know, thank you.

**Comment by DukhUma** [comment]
Came here to second point number 4. Many extensions do not benefit much from instant access and only serve to clutter the navbar and crowd out the URL. Much better option would be to choose which are made visible and hide the rest in menu. Such a shape we can't easily edit UI elements on the fly on Android as we can desktop.
